core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -152,8 +152,8 @@ class LogCleaner(val config: CleanerConfig,
    * For testing, a way to know when work has completed. This method blocks until the
    * cleaner has processed up to the given offset on the specified topic/partition
    */
-  def awaitCleaned(topic: String, part: Int, offset: Long, timeout: Long = 30000L): Unit = {
-    while(!cleanerManager.allCleanerCheckpoints.contains(TopicAndPartition(topic, part)))
+  def awaitCleaned(topic: String, part: Int, offset: Long): Unit = {
```

Comment:
@ijuma Yeah I know we have default value of 30000L for timeout. My point is that, since it is not explicitly specified in the invocation is probably means users don't care about it. It is important that, if we enable this timeout, existing tests won't fail due to low timeout value.

I am not sure.. but can you explain why 30000 ms is a good default value for timeout? Note that log cleaner's default backoffMs is 15000 ms.

Alternatively, how about I set timeout to be LONG.MAX by default? This won't accidentally fail existing test while still allow developers to use timeout when they want.